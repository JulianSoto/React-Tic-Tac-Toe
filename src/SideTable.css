.side-table {
	position: relative;
	width: 150px;
	height: 150px;
	margin-left: 50px;
}

.status {
	position: absolute;
	bottom: 0;
	left: 0;
	width: 100%;
}

.home {
	width: 100%;
}

.game-mode-option {
	width: 100%;
}

.play-again {
	width: 100%;
}

.scoreboard {
	display: block;
	margin-top: 10px;
	font-size: 1.5em;
	color: #444;
}