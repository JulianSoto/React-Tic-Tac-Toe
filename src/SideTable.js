//side table
import React, {Component} from 'react';
import {HUMAN_VS_HUMAN, HUMAN_VS_COMPUTER} from './globalConstants.js';
import Status from './StatusMessage.js';

export default class SideTable extends Component {
	constructor(props){
		super(props);
	}

	render(){
		if (this.props.gameMode === null){
			return (
				<div>
					<button
						onClick={() => this.props.gameModeSelectionHandler(HUMAN_VS_HUMAN)}
					>
						Human vs human
					</button>
					<button
						onClick={() => this.props.gameModeSelectionHandler(HUMAN_VS_COMPUTER)}
					>
						Human vs computer
					</button>
					<span>Choose a game mode</span>
				</div>
			)
		}

		if (!this.props.players){
			return (
				<div>
					<button onClick={() => this.props.resetGameState()}>Home</button>
					<button onClick={() => this.props.selectX()}>X</button>
					<button onClick={() => this.props.selectO()}>O</button>
					<span>Select X or O</span>
				</div>
			)
		}

		return (
			<div>
				<button onClick={() => this.props.resetGameState()}>Home</button>
				<button>Reset</button>
				<Status
					winner={this.props.winner}
					players={this.props.players}
					isDraw={this.props.isDraw}
					gameMode={this.props.gameMode}
					turn={this.props.turn}
				/>
				<span>
					{`${this.props.scoreboard[0]} - ${this.props.scoreboard[1]}`}
				</span>
			</div>
		)
	}
}