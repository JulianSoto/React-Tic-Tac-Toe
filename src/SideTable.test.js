import React from 'react';
import { shallow } from 'enzyme';
import SideTable from './SideTable.js';

describe('SideTable', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<SideTable/>);
		expect(wrapper).toMatchSnapshot();
	})
})