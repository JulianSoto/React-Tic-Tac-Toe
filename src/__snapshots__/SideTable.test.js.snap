// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideTable should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideTable />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
        >
          Home
        </button>,
        <button
          onClick={[Function]}
        >
          X
        </button>,
        <button
          onClick={[Function]}
        >
          O
        </button>,
        <span>
          Select X or O
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Home",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Home",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "O",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "O",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select X or O",
        },
        "ref": null,
        "rendered": "Select X or O",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
          >
            Home
          </button>,
          <button
            onClick={[Function]}
          >
            X
          </button>,
          <button
            onClick={[Function]}
          >
            O
          </button>,
          <span>
            Select X or O
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Home",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "O",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "O",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select X or O",
          },
          "ref": null,
          "rendered": "Select X or O",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
